model/task_queue_service_dependency.go

```diff
@@ -511,5 +511,14 @@ func checkUnmarkedBlockingTasks(t *task.Task, dependencyCaches map[string]task.T
 		}
 	}
 
+	// also update the display task status in case it is out of date
+	if t.IsPartOfDisplay() {
```

Comment:
Could this check if len(blockingTask)s > 0 or len(blockingDeactivatedTasks) > 0? I imagine that if neither of those are true then the display task doesn't need refreshing.

Reply:
I believe we also need to refresh the display task if there are no blocked execution tasks, in case it was blocked before but is not blocked now